llama/generation.py

```diff
@@ -56,14 +73,17 @@ def generate(
         params = self.model.params
         assert bsz <= params.max_batch_size, (bsz, params.max_batch_size)
 
-        prompt_tokens = [self.tokenizer.encode(x, bos=True, eos=False) for x in prompts]
+        bos = not USE_CUDA  # The supplied t5 tokenizer doesn't have bos. CUDA will error out but xla won't.
```

Comment:
Sorry for the late comment. Would this affect the users that try to use the LLaMA checkpoint and tokenizer?

Reply:
Good point. Yea, it would.  Is there a way to tell whether this is a T5 tokenizer or LLaMA? I can do a follow up.

Reply:
Maybe a `try...except..` to catch and fall back to `False`?

Reply:
How? The error only manifest in the later embedding access code: https://github.com/pytorch-tpu/llama/blob/main/llama/model.py#L225. The CUDA kernel crashes...